src/compiler/checker.ts

```diff
@@ -17815,6 +17815,8 @@ namespace ts {
             const t = target.flags;
             if (t & TypeFlags.AnyOrUnknown || s & TypeFlags.Never || source === wildcardType) return true;
             if (t & TypeFlags.Never) return false;
+            // Intersection types that would be reduced to never may be passed as `target`, thus returning early.
+            if (t & TypeFlags.Intersection) return false;

```

Comment:
This isn't quite the right fix, I will put something up in a PR.

Reply:
Thanks for the review. Besides my fix not allowing empty array literals assigned to `never[]`, is it not "quite the right fix" because it fails to catch cases where `target` is a union of empty intersections that also reduce to `never`? A bit of explanation or even pointers would be appreciated if you don't mind!

Reply:
Yes, one issue is that both unions and intersections can potentially reduce to `never`, so both need to be excluded when `--strictNullChecks` is disabled. But another issue is that `any` should be assignable to anything, which is handled by a test later in the body of `isSimpleTypeRelatedTo`. So the code can't just unconditionally return `false` here.

Reply:
Thanks! I'll think about those logic more carefully next time.